Sources/MongoSwift/APM.swift

```diff
@@ -55,23 +152,35 @@ public struct CommandStartedEvent: MongoCommandEvent, InitializableFromOpaquePoi
     /// The address of the server the command was run against.
     public let serverAddress: Address
 
-    /// Initializes a CommandStartedEvent from an OpaquePointer to a mongoc_apm_command_started_t
-    fileprivate init(_ event: OpaquePointer) {
+    fileprivate init(mongocEvent: MongocCommandStartedEvent) {
         // we have to copy because libmongoc owns the pointer.
-        self.command = Document(copying: mongoc_apm_command_started_get_command(event))
-        self.databaseName = String(cString: mongoc_apm_command_started_get_database_name(event))
-        self.commandName = String(cString: mongoc_apm_command_started_get_command_name(event))
-        self.requestId = mongoc_apm_command_started_get_request_id(event)
-        self.operationId = mongoc_apm_command_started_get_operation_id(event)
-        self.serverAddress = Address(mongoc_apm_command_started_get_host(event))
+        self.command = Document(copying: mongoc_apm_command_started_get_command(mongocEvent.ptr))
```

Comment:
Because the context function needed to be called elsewhere and virtually, I abstracted it to the protocol. The rest of these are only called here, so I figured it would be overkill to implement them all on the `MongocCommandStartedEvent` type (and the same thing goes for all the other mongoc event type wrappers)